test/integration/gpu_rays.cc

```diff
@@ -459,6 +463,173 @@ void GpuRaysTest::LaserVertical(const std::string &_renderEngine)
+                                   ignition::math::Quaterniond::Identity);
+  VisualPtr visualBox1 = scene->CreateVisual("UnitBox1");
+  visualBox1->AddGeometry(scene->CreateBox());
+  visualBox1->SetWorldPosition(box01Pose.Pos());
+  visualBox1->SetWorldRotation(box01Pose.Rot());
+  root->AddChild(visualBox1);
+
+  // box on the right of the first gpu rays caster
+  ignition::math::Pose3d box02Pose(ignition::math::Vector3d(0, -5, 0.5),
+                                   ignition::math::Quaterniond::Identity);
+  VisualPtr visualBox2 = scene->CreateVisual("UnitBox2");
+  visualBox2->AddGeometry(scene->CreateBox());
+  visualBox2->SetWorldPosition(box02Pose.Pos());
+  visualBox2->SetWorldRotation(box02Pose.Rot());
+  root->AddChild(visualBox2);
+
+  // box on the left of the rays caster 1 but out of range
+  ignition::math::Pose3d box03Pose(
+      ignition::math::Vector3d(0, maxRange + 1, 0.5),
+      ignition::math::Quaterniond::Identity);
+  VisualPtr visualBox3 = scene->CreateVisual("UnitBox3");
+  visualBox3->AddGeometry(scene->CreateBox());
+  visualBox3->SetWorldPosition(box03Pose.Pos());
+  visualBox3->SetWorldRotation(box03Pose.Rot());
+  root->AddChild(visualBox3);
+
+  // create particle emitter between sensor and box in the center
+  ignition::math::Vector3d particlePosition(1.0, 0, 0);
+  ignition::math::Vector3d particleSize(0.2, 0.2, 0.2);
+  ignition::rendering::ParticleEmitterPtr emitter =
+      scene->CreateParticleEmitter();
+  emitter->SetLocalPosition(particlePosition);
+  emitter->SetParticleSize(particleSize);
+  emitter->SetRate(100);
+  emitter->SetLifetime(2);
+  emitter->SetVelocityRange(0.1, 0.1);
+  emitter->SetScaleRate(0.2);
+  emitter->SetColorRange(ignition::math::Color::Red,
+      ignition::math::Color::Black);
+  emitter->SetEmitting(true);
+  root->AddChild(emitter);
+
+  // Verify rays caster 1 range readings
+  // listen to new gpu rays frames
+  unsigned int channels = gpuRays->Channels();
+  float *scan = new float[hRayCount * vRayCount * channels];
+  common::ConnectionPtr c =
+    gpuRays->ConnectNewGpuRaysFrame(
+        std::bind(&::OnNewGpuRaysFrame, scan,
+          std::placeholders::_1, std::placeholders::_2, std::placeholders::_3,
+          std::placeholders::_4, std::placeholders::_5));
+
+
+  int mid = static_cast<int>(hRayCount/2) * channels;
+  int last = (hRayCount - 1) * channels;
+  double unitBoxSize = 1.0;
+  double expectedRangeAtMidPointBox1 = abs(box01Pose.Pos().X()) - unitBoxSize/2;
+  double expectedRangeAtMidPointBox2 = abs(box02Pose.Pos().Y()) - unitBoxSize/2;
+
+  // set a larger tol for particle range
+  double laserNoiseTol = particleSize.X();
+  double expectedParticleRange = particlePosition.X();
+
+  // update 100 frames. There should be a descent chance that we will see both
+  // a particle hit and miss in the readings returned by the sensor
+  unsigned particleHitCount = 0u;
+  unsigned particleMissCount = 0u;
```

Comment:
```suggestion
  unsigned int particleHitCount = 0u;
  unsigned int particleMissCount = 0u;
```
I think `unsigned` and `unsigned int` resolve to the same thing, but it'd be good to use `unsigned int` for clarity.

Reply:
4108632